test/entity.ts

```diff
@@ -744,32 +744,77 @@ describe('entity', () => {
       assert.deepStrictEqual(entity.encodeValue(value), expectedValueProto);
     });
 
-    it('should encode an int', () => {
+    it('should encode an int', done => {
       const value = 8;
 
       const expectedValueProto = {
         integerValue: value,
       };
 
+      const property = 'undefined';
+      const expectedWarning =
+        "TypeCastWarning: the value for '" +
+        property +
+        "' property is a JavaScript Number.\n" +
+        "Use 'Datastore.int(<integer_value_as_string>)' or " +
+        "'Datastore.double(<double_value_as_string>)' to preserve consistent " +
+        'Datastore types in your database.';
+      process.once('warning', warning => {
+        assert.strictEqual(warning.message, expectedWarning);
+        done();
+      });
+
+      entity.Int = function (value_: {}) {
+        assert.strictEqual(value_, value);
+        this.value = value_;
+      };
+
+      assert.deepStrictEqual(entity.encodeValue(value), expectedValueProto);
+    });
+
+    it('should encode an int but only warn once', done => {
+      const value = 8;
+
+      const expectedValueProto = {
+        integerValue: value,
+      };
+
+      const property = 'undefined';
+      const expectedWarning =
+        "TypeCastWarning: the value for '" +
+        property +
+        "' property is a JavaScript Number.\n" +
+        "Use 'Datastore.int(<integer_value_as_string>)' or " +
+        "'Datastore.double(<double_value_as_string>)' to preserve consistent " +
+        'Datastore types in your database.';
+      process.once('warning', warning => {
+        assert.strictEqual(warning.message, expectedWarning);
+        done();
+      });
+
       entity.Int = function (value_: {}) {
```

Comment:
Ah, I see, this was already in there. Anyhow, I think my earlier comment is still meaningful to check.